src/test/java/org/folio/rest/impl/GOBIIntegrationServiceResourceImplTest.java

```diff
@@ -491,11 +492,85 @@ public final void testPostContentWithValidOkapiToken(TestContext context) throws
+    logger.info("Begin: Testing for falling back to the unspecified material id, if a non existent code is sent");
+
+    final Async asyncLocal = context.async();
+
+    final String body = getMockData(POLISTEDELECTRONICSERIALPATH);
+
+    final GobiResponse order = RestAssured
+      .given()
+       .header(TOKENHEADER)
+       .header(URLHEADER)
+       .header(TENANTHEADER)
+       .header(CONTENTTYPEHEADERXML)
+      .body(body)
+      .when()
+      .post(ORDERSPATH)
+       .then()
+         .statusCode(201)
+         .contentType(ContentType.XML)
+         .extract()
+         .body()
+         .as(GobiResponse.class, ObjectMapperType.JAXB);
+
+    context.assertNotNull(order.getPoLineNumber());
+
+    List<JsonObject> configEntries = MockServer.serverRqRs.get(MATERIAL_TYPES, HttpMethod.GET);
+    //2 calls must be made to material types end point, once for non existent and other for unspecified
+    assertEquals(2, configEntries.size());
+
+    List<JsonObject> postedOrder = MockServer.serverRqRs.get(PURCHASEORDER, HttpMethod.POST);
+    CompositePurchaseOrder po = postedOrder.get(0).mapTo(CompositePurchaseOrder.class);
+    assertEquals(UNSPECIFIED_MATERIAL_TYPE_ID, po.getCompositePoLines().get(0).getDetails().getMaterialTypes().get(0));
+
+    asyncLocal.complete();
+
+    logger.info("End: Testing for falling back to the unspecified material id, if a non existent code is sent");
+  }
+
+  @Test
+  public final void testPostGobiOrdersFallBackLocation(TestContext context) throws Exception {
+    logger.info("Begin: Testing for falling back to the first location id, if a non existent code is sent");
+
+    final Async asyncLocal = context.async();
+
+    final String body = getMockData(POLISTEDELECTRONICSERIALPATH);
+
+    final GobiResponse order = RestAssured
+      .given()
+       .header(TOKENHEADER)
+       .header(URLHEADER)
+       .header(TENANTHEADER)
+       .header(CONTENTTYPEHEADERXML)
+      .body(body)
+      .when()
+      .post(ORDERSPATH)
+       .then()
+        .statusCode(201)
+        .contentType(ContentType.XML)
+        .extract()
+        .body()
+        .as(GobiResponse.class, ObjectMapperType.JAXB);
+
+    context.assertNotNull(order.getPoLineNumber());
+
+    List<JsonObject> configEntries = MockServer.serverRqRs.get(LOCATION, HttpMethod.GET);
+    // 2 calls must be made to location end point
+    assertEquals(2, configEntries.size());
+
+    asyncLocal.complete();
+
+    logger.info("End: Testing for falling back to tthe first location id, if a non existent code is sent");
+  }
+
   public static class MockServer {
 
-    private static final String LOCATION = "LOCATION";
-    private static final String MATERIAL_TYPES = "MATERIAL-TYPES";
-    private static final String VENDOR = "VENDOR";
+    private static final String NAME = "name";
+    private static final String ID = "id";
+    private static final String TOTAL_RECORDS = "total_records";
```

Comment:
Shouldn't this be `totalRecords` or we are yet to sync with mod-orders in terms of camelCase naming convention?